hudi-common/src/main/java/org/apache/hudi/common/table/read/InputSplit.java

```diff
@@ -81,4 +89,13 @@ public long getLength() {
   public boolean isParquetBaseFile() {
     return baseFileOption.map(baseFile -> HoodieFileFormat.fromFileExtension(baseFile.getStoragePath().getFileExtension()) == HoodieFileFormat.PARQUET).orElse(false);
   }
+
+  public boolean hasNoRecordsToMerge() {
+    return this.logFiles.isEmpty() && recordIterator.isEmpty();
+  }
+
+  public Option<Iterator<Pair<Serializable, BufferedRecord>>> getRecordIterator() {
+    ValidationUtils.checkArgument(recordIterator.isPresent(), "Record iterator is not initialized");
```

Comment:
Let's add the check in `getLogFiles()` that record iterator should not be set for using log files.

Reply:
fixed